src/primitives/field.rs

```diff
@@ -362,6 +364,19 @@ macro_rules! impl_ops_for_fe {
                 self._neg()
             }
         }
+
+        // sum
+        impl core::iter::Sum for $op {
+            fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+                iter.fold(crate::primitives::Field::ZERO, |i, acc| i + acc)
+            }
+        }
+
+        impl<'s> core::iter::Sum<&'s Self> for $op {
+            fn sum<I: Iterator<Item = &'s Self>>(iter: I) -> Self {
+                iter.fold(crate::primitives::Field::ZERO, |i, acc| i + acc)
+            }
+        }
```

Comment:
Is there any technical reason you chose to not add a `_sum` trait method to the `Field` trait and call it here like the other code does? If not I rekon we should keep it uniform.

Reply:
The reason the `_add` etc methods exist is so that I could implement the 4 different `Add` traits with a single macro call.

There is only one `Sum` trait to implement and it can be implemented in terms of `_add`. I don't see what the value of having a `_sum` method would be. Would I insist that implementors implement it?

Reply:
If anything I should delete the `_add` methods and improve the macro. I may do this in a followup.

Reply:
No sweat, cheers.

Reply:
I read the code a bit more, I thought the `_foo` methods were just to make `impl_ops_for_fe` less duplicated, my mistake.

Reply:
Thanks